core/clients/coredataclients/coredata-clients.go

```diff
@@ -72,6 +77,19 @@ func decodeValueDescriptorSlice(resp *http.Response) ([]models.ValueDescriptor,
 	return dSlice, err
 }
 
+// Helper method to decode a device and return the device
+func (v *ValueDescriptorClient) decodeValueDescriptor(resp *http.Response) (models.ValueDescriptor, error) {
+	dec := json.NewDecoder(resp.Body)
+	vdr := models.ValueDescriptor{}
+
+	err := dec.Decode(&vdr)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Printing on the stdout should be avoided. Use logs instead.

Reply:
You can also retun immediately here, with `return nil, err` - no need returning a struct below, when we have an error.